src/antelope/chains/evm/telos-evm-testnet/index.ts

```diff
@@ -59,8 +61,14 @@ export default class TelosEVMTestnet extends EVMChainSettings {
         return { ...TOKEN, tokenId: this.constructTokenId(TOKEN) } as EvmToken;
     }
 
-    getUsdPrice(): Promise<number> {
-        return api.getCoingeckoUsdPrice('telos');
+    async getUsdPrice(): Promise<number> {
+        if (this.hasIndexSupport()) {
+            const nativeTokenSymbol = this.getSystemToken().symbol;
+            const fiatCode = useUserStore().fiatCurrency;
+            return getFiatPriceFromIndexer(nativeTokenSymbol, fiatCode, this.indexer);
+        } else {
+            return api.getCoingeckoUsdPrice('telos');
```

Comment:
ensure balance is always coming from indexer when possible-- we were still getting price from coingecko in some cases because of this function